agent/proto/agentpb/agent.proto

```diff
@@ -79,20 +79,24 @@ message NodeStatus {
 }
 
 message Probe {
-  // Name of the checker that generated the probe.
+  // Checker is the name of the checker that generated the probe
   string checker = 1;
-  // Optional detail specific to the checker
+  // Detail is the optional detail specific to the checker
   string detail = 2;
-  // Optional code specific to a checker (i.e. HTTP status code)
+  // Code is the optional code specific to a checker (i.e. HTTP status code)
   string code = 3;
   enum Type {
     Unknown = 0;
     Running = 1;
     Failed = 2;
     Terminated = 3;
   }
+  // Status is the result of the probe
   Type status = 4;
+  // Error is the probe-specific error message
   string error = 5;
+  // CheckerData is a free-form data specific to the checker
+  bytes checker_data = 6;
```

Comment:
I think it's just a style difference, but I don't think I'm a huge fan of embedding arbitrary data like this by encoding a structure to json. I realize it's only important to specific probes, but to me it seems far more straightforward to extend the message and generate compile time failures if there are breaking changes.

Reply:
Yeah, thought about it too: I didn't want to end up with the mess where this struct contains all possible fields.